### PR TITLE
New msprime calling convention

### PR DESCRIPTION
**Pull request by hyanwong**

### CODE REVIEW

**Comment by jeromekelleher** [comment]
I don't think we should change the code until msprime 1 is released (or at least we have a alpha/beta). Currently this isn't being run as part of CI, so it's clearly not being tested.

**Comment by hyanwong** [comment]
ISTR that I changed it because I was getting warnings when running `nosetests` when I had installed a GitHub version of msprime. But let's leave this open until msprime 1?

**Comment by jeromekelleher** [comment]
> ISTR that I changed it because I was getting warnings when running nosetests when I had installed a GitHub version of msprime.

Yes, but we can't change the code for everyone until msprime is released. Probably best to revert to the released version of msprime while doing tsinfer dev.

**Comment by hyanwong** [comment]
Just to follow up, tests are indeed failing on master if a bleeding edge msprime is installed:

```
test_multiple_recombinations (test_evaluation.TestInsertPerfectMutations) ... /usr/local/anaconda3/lib/python3.8/site-packages/msprime/intervals.py:187: UserWarning: num_loci is no longer supported and should not be used.
  warnings.warn("num_loci is no longer supported and should not be used.")
FAIL
```

**Comment by jeromekelleher** [comment]
> Just to follow up, tests are indeed failing on master if a bleeding edge msprime is installed:

Yes - which is why I said to stick with the released version.

**Comment by hyanwong** [comment]
Yes, sorry - messages crossed. Still, it will be annoying when tsinfer unit tests start failing because CI starts using msprime 1.0. So I was just sorting it out so that everything would carry on working.

**Comment by hyanwong** [comment]
I'm just starting to use the msprime alpha for tsinfer testing. This squashes a few errors, but I'm still getting errors like 

```
>       assert len(tables.populations.metadata) == 0
E       AssertionError

tests/tsutil.py:48: AssertionError
```
 And indeed at

https://github.com/tskit-dev/tsinfer/blob/25312f75c3ecbe6de62a5380cceb4ac2df08b2fb/tests/tsutil.py#L42

it says "This will probably fail when msprime 1.0 comes along, but we can fix it then.".

I'm not clear how the permissive schemas etc conflict with msprime now, perhaps @benjeffery or @jeromekelleher can advise?

**Comment by jeromekelleher** [comment]
> I'm not clear how the permissive schemas etc conflict with msprime now, perhaps @benjeffery or @jeromekelleher can advise?

Probably some details to be worked out, I wouldn't know without digging into it. Probably not worth messing with until msprime 1.0 is fully packaged though.

**Comment by jeromekelleher** [comment]
Can you bring this up to date please @hyanwong?

**Comment by hyanwong** [comment]
Ah, I remember the problem. Weirdly, even with the changes in here to remove mutation times, the new msprime is causing the following tests to fail:

```
FAILED tests/test_formats.py::TestAncestorData::test_ancestors_truncated_length - assert 431 == 213
FAILED tests/test_inference.py::TestMatchSamples::test_partial_samples - assert False
FAILED tests/test_inference.py::TestPathCompressionAncestorsPyEngine::test_simulation_with_error - assert 0 > 0
FAILED tests/test_inference.py::TestPathCompressionAncestorsCEngine::test_simulation_with_error - assert 0 > 0
```

I'll have a dig, but it's unclear to me what the problem is.

File: tests/test_evaluation.py [hunk 1]
```diff
@@ -397,15 +398,15 @@ def test_large_smc(self):
...
         try:
-            recomb_map = msprime.RecombinationMap.uniform_map(
-                length=10, rate=10, discrete=True
+            ts = msprime.sim_ancestry(
+                5, sequence_length=10, recombination_rate=10, random_seed=1
             )
-        except TypeError:
-            # Fallback for older versions of msprime, which use num_loci
-            recomb_map = msprime.RecombinationMap.uniform_map(
+        except AttributeError:
```
Comment (jeromekelleher) [inline tests/test_evaluation.py:404]: No point in keeping this now, we should use either sim_ancestry or simulate

File: tests/tsutil.py [hunk 2]
```diff
@@ -28,6 +28,19 @@
...
 
+def mark_mutation_times_unknown(ts):
+    """
+    Recent msprime versions provide times for mutations, which cannot be estimated by
+    tsinfer
+    """
+    tables = ts.dump_tables()
+    try:
+        tables.mutations.time = np.full_like(tables.mutations.time, tskit.UNKNOWN_TIME)
+    except AttributeError:
```
Comment (jeromekelleher) [inline tests/tsutil.py:39]: We're not interested in supporting older versions of msprime. We can add msprime>=1.0 to requirements.txt

**Review by jeromekelleher — COMMENTED** [review]
Looks good, thanks. Can you update the requirements.txt files to require msprime > 1 and clear out the conditional stuff please?

**Comment by hyanwong** [comment]
> Looks good, thanks. Can you update the requirements.txt files to require msprime > 1 and clear out the conditional stuff please?

Removed conditionals for old msprime and changed `requirements/development.txt` and the CI `requirements.txt` files where it was mentioned. Guess that's OK.

**Comment by hyanwong** [comment]
CI now failing with 

```
ImportError: Legacy formats require h5py. Install via `pip install h5py` or `conda install h5py`
```

Any idea why that's suddenly appeared @jeromekelleher ? Do the requirements need changing?

**Comment by jeromekelleher** [comment]
> ImportError: Legacy formats require h5py. Install via `pip install h5py` or `conda install h5py`

Not sure why we'd be reading in legacy files here @hyanwong, but it might simplest to put h5py into the various requirements.txt files for CI and development. (Either that or figure out where we're reading in hdf5 files and get rid of them - these are super old now and irrelevant for tsinfer.)

**Comment by hyanwong** [comment]
> > ImportError: Legacy formats require h5py. Install via `pip install h5py` or `conda install h5py`
> 
> Not sure why we'd be reading in legacy files here @hyanwong, but it might simplest to put h5py into the various requirements.txt files for CI and development. (Either that or figure out where we're reading in hdf5 files and get rid of them - these are super old now and irrelevant for tsinfer.)

It's in `test_infer_bad_file` - I suspect we fall back to trying to load old formats.

**Comment by jeromekelleher** [comment]
> It's in test_infer_bad_file - I suspect we fall back to trying to load old formats.

Just nuke the hdf5 files, they're irrelevant now

**Comment by hyanwong** [comment]
There aren't any hdf5 files. We try to load a random file (e.g. a completely empty one), and fall back to using tskit if we can't recognise it. Tskit doesn't like the look of the file, so falls back to checking if it's hdf5 format. Bang.

**Comment by jeromekelleher** [comment]
Ah - well, just add in the requirements then.

File: .github/workflows/wheels.yml [hunk 3]
```diff
@@ -145,7 +145,7 @@ jobs:
     runs-on: macos-latest
     strategy:
       matrix:
-        python: [3.6, 3.7, 3.8]
+        python: [3.7, 3.8]
```
Comment (benjeffery) [inline .github/workflows/wheels.yml:148]: 3.9 was removed when we had issues with it - I think it can be reinstated now?

**Review by benjeffery — APPROVED** [review]
LGTM - one question about wheel building.

File: .github/workflows/wheels.yml (hunk 3 above)
Comment (hyanwong) [reply .github/workflows/wheels.yml]: Good point. Just added a commit to reinstate, seems fine.

**Pull request merged**